sourmash/signature_json.py

```diff
@@ -235,29 +235,25 @@ def load_signatures_json(data, ksize=None, ignore_md5sum=True, ijson=ijson):
         notify('\r...sig loading {:,}', n, flush=True)
 
 
-def get_top_records(siglist):
-    # @CTB I think get_top_records can be removed? It doesn't serve much
-    # of a purpose, and (in some old versions of Python at least) it
-    # may randomize the order of signatures in siglist, which is an
-    # undesirable side effect.
+def add_meta_save(siglist):
+    """ Convert one signature into a JSON dict
+    - siglist: sequence of SourmashSignature objects
+    - index: index of siglist to save
+    """
+    from .signature import SIGNATURE_VERSION
+    records = []
     top_records = {}
+
     for sig in siglist:
         name, filename, sketch = sig._save()
         k = (name, filename)
         x = top_records.get(k, [])
         x.append(sketch)
         top_records[k] = x
 
-    return top_records
-
+    if not top_records:
+        return ""
 
-def add_meta_save(top_records):
-    """ Convert one signature into a JSON dict
-    - siglist: sequence of SourmashSignature objects
-    - index: index of siglist to save
-    """
-    from .signature import SIGNATURE_VERSION
-    records = []
     for (name, filename), sketches in top_records.items():
```

Comment:
Why use a dictionary here for `top_records`, and not a list?

Reply:
To accumulate all the molecule type and ksizes for a cell barcode into one signatures list. 
FWIW, I didn't edit this

Reply:
ok, I just misunderstood the code. good grief :). thanks, looks good.